src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -120,18 +120,18 @@ public static ReadOnlySpan<char> Trim(this ReadOnlySpan<char> span, ReadOnlySpan
         public static ReadOnlySpan<char> TrimStart(this ReadOnlySpan<char> span, ReadOnlySpan<char> trimChars)
         {
             int start = 0;
-            for (; start < span.Length; start++)
+            while (start < span.Length)
             {
                 int i = 0;
                 for (; i < trimChars.Length; i++)
                 {
                     if (span[start] == trimChars[i])
-                        goto Done;
+                        goto Next;
                 }
-                if (i == trimChars.Length)
-                    break;
+                break;
+            Next:
+                start++;
```

Comment:
Does the JIT bounds check elimination pick up this pattern correctly?

I do not think there was anything wrong with for-loop. `for (int start = 0; start < span.Length; start++)` should work fine.

Reply:
I can't label a line for the `goto` which  just is essentially empty (i.e. the for loop brace bracket).

Reply:
Add semicolon at the end of the line.

Updated: https://github.com/dotnet/corefx/pull/26560#discussion_r163719355

Reply:
> Does the JIT bounds check elimination pick up this pattern correctly?

I don't see any extra compares in the disassembly.

```C#
for (int i = 0; i < trimChars.Length; i++)
{
    if (span[end] == trimChars[i])
        goto Next;
}
break;
```
```text
00007fff`7a873118 7e2c            jle  to (00007fff`7a873146)
00007fff`7a87311a 33f6            xor     esi,esi
00007fff`7a87311c 4585c0          test    r8d,r8d
00007fff`7a87311f 7e25            jle      to 00007fff`7a873146)
00007fff`7a873121 4963c3          movsxd  rax,r11d
00007fff`7a873124 410fb70442      movzx   eax,word ptr [r10+rax*2]
00007fff`7a873129 4863fe          movsxd  rdi,esi
00007fff`7a87312c 410fb73c79      movzx   edi,word ptr [r9+rdi*2]
00007fff`7a873131 3bf8            cmp     edi,eax
00007fff`7a873133 7409            je       to (00007fff`7a87313e)
00007fff`7a873135 ffc6            inc     esi
00007fff`7a873137 413bf0          cmp     esi,r8d
00007fff`7a87313a 7ced            jl      to (00007fff`7a873129)
```